tests/testthat/test_format_dta.R

```diff
@@ -9,7 +9,18 @@ test_that("Export to Stata", {
 test_that("Import from Stata (read_dta)", {
     expect_true(is.data.frame(import("mtcars.dta", haven = TRUE)))
     # arguments ignored
-    expect_warning(is.data.frame(import("mtcars.dta", haven = TRUE, extraneous.argument = TRUE)))
+    expect_warning(import("mtcars.dta", haven = TRUE, extraneous.argument = TRUE))
+    # tell arg_reconcile() to not show warnings
+    expect_silent(import("mtcars.dta", haven = TRUE,
+                         extraneous.argument = TRUE, .warn = FALSE))
+})
+
+test_that("Import from Stata with extended Haven features (read_dta)", {
```

Comment:
This is the part that demonstrates how this approach can generically address problems of which #248 is an example.